src/CommunityToolkit.Maui/Behaviors/ICommunityToolkitBehavior.shared.cs

```diff
@@ -21,7 +21,12 @@ internal bool TrySetBindingContextToAttachedViewBindingContext()
 			throw new InvalidOperationException($"{nameof(ICommunityToolkitBehavior<TView>)} can only be used for a {nameof(Behavior)}");
 		}
 
-		if (behavior.IsSet(BindableObject.BindingContextProperty) || View is null)
```

Comment:
What will happen in scenarios where a developer explicitly sets their own BindingContext for the behavior? I think this could would overwrite it with the BindingContext from the View?

Reply:
Perhaps the safer change is to remove the binding altogether and subscribe to the BindingContext changed event on the View and assign to the Behavior?

Reply:
Not being able to re-assing the `BindingContext` will make all Behaviors un-usable on recycling/virtualization scenarios.

What do you think the correct approach would be?

Reply:
I have changed the `OnAttachedTo` method in `BaseBehavior` to look as follows:

```csharp
protected override void OnAttachedTo(TView bindable)
{
	base.OnAttachedTo(bindable);

	bindable.PropertyChanging += (sender, args) =>
	{
		var currentViewBindingContext = ((TView)sender).BindingContext;
			
		if (args.PropertyName == nameof(BindingContext) && 
		    ReferenceEquals(this.BindingContext, currentViewBindingContext))
		{
			this.BindingContext = null;
		}
	};
		
	bindable.PropertyChanged += (sender, args) =>
	{
		if (args.PropertyName == nameof(BindingContext) &&
		    this.BindingContext is null)
		{
			this.BindingContext = ((TView?)sender)?.BindingContext;				
		}
    };
}
```

It might not be the prettiest but I believe it should work. What does everyone think? Any ideas on how to make it nicer?

Reply:
@bijington what's the difference between using event handlers and the interface? At the end we will update the BindingContext value

Reply:
Unless I'm missing something my code should only update the BindingContext if the behavior previously used the value from the View. This means we wouldn't override the value of a user supplied a BindingContext.

Reply:
Oh I see now, yeah you're right about the behavior on your code... But from my memory there isn't anything on .NET MAUI that prevents the BindingContext to be changed... it shouldn't be a `readonly` or `immutable`  value. So I'm curious, why this implementation was used

Reply:
I believe the reasoning is that we only inherit the BindingContext from the attached control if the developer hasn't explicitly set the BindingContext themselves. I have to be honest and say I haven't checked the order things happen then and when the BindingContext becomes not null

Reply:
I have reverted the change and added in two examples from @tranb3r example repo. Perhaps @tranb3r you can test this sample and let us know how it compares to your expectation. The code works as I expect on iOS/macOS in our sample app

Reply:
@bijington will take a look asap. Did you add the failing case to the sample app? Asking to make sure  Ican use it during development

Reply:
Yes there are now 2 CollectionViews in the sample app on the TouchBehaviorPage

Reply:
> I have reverted the change and added in two examples from @tranb3r example repo. Perhaps @tranb3r you can test this sample and let us know how it compares to your expectation. The code works as I expect on iOS/macOS in our sample app

These are indeed the same cases as in my repro. Not working on android, unless using 8.0.0.

Reply:
@tranb3r thank you for confirming.

@tranb3r and @pictos the changes that I have just pushed appear to solve the issue on Android based on my testing.

Reply:
Sorry to ask again, but I'm still wondering if this BindingContext auto-set feature (introduced in 8.0.1) is really required?
It seems to me that 8.0.0 was fine. If the goal was only to simplify xaml code, I'm not sure it's worth it, considering the difficulty you have to figure out how to fix this regression.
I'm probably missing something, but could you please explain which bug did you try to fix in 8.0.1? Or at least post a code sample for the original issue?

Reply:
I recall there was a lot of developers reporting an issue that their bindings didn't work. @brminnick Is this correct?

Reply:
I suppose their bindings didn't work because they were not setting the BindingContext manually.